extensions/pkg/controller/worker/genericactuator/actuator_delete.go

```diff
@@ -99,6 +99,11 @@ func (a *genericActuator) Delete(ctx context.Context, worker *extensionsv1alpha1
 		return errors.Wrapf(err, "failed deleting machine-controller-manager")
 	}
 
+	// Cleanup machine dependencies.
```

Comment:
Does it make sense to redeploy `DeployMachineDependecies` (https://github.com/gardener/gardener/blob/11052936434e07373e7774be18ea98830575cdc2/extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go#L73-L76) during deletion as well? 

We do similar operations for the deletion of MachineClasses here - https://github.com/gardener/gardener/blob/11052936434e07373e7774be18ea98830575cdc2/extensions/pkg/controller/worker/genericactuator/actuator_delete.go#L60-L65.  

I am just wondering if something like this would help to update the objects before the actual cleanup in a deletion loop. In the case of machine classes, we would be updating the cloud credentials on the secrets. I am only partially able to imagine the flow in the machine dependencies case, just wondering if something like this makes sense for machine dependencies as well? @dkistner you would have a better idea at this?

Reply:
Probably it doesn't make sense. The reason for redeploying the machine classes is to update the machine class secrets. But the machine dependencies can be seen as infrastructure, and we also don't redeploy the infrastructure before deletion.

Reply:
Okay yes. Even I am unable to come up with scenarios that would help to implement it this way. I agree with you, I am going to resolve this for now.